src/net.cpp

```diff
@@ -2468,9 +2468,24 @@ void CConnman::AddNewAddresses(const std::vector<CAddress>& vAddr, const CAddres
     addrman.Add(vAddr, addrFrom, nTimePenalty);
 }
 
-std::vector<CAddress> CConnman::GetAddresses()
+std::vector<CAddress> CConnman::GetAddresses(AddrCacheNetwork use_cache)
 {
-    return addrman.GetAddr();
+    if (use_cache == ADDR_CACHE_NONE) {
+        return addrman.GetAddr();
+    } else {
+        CachedAddrResponse* cache;
+        if (use_cache == ADDR_CACHE_REGULAR) {
+            cache = &m_cached_addr_response_regular;
+        } else {
+            cache = &m_cached_addr_response_other;
+        }
+        const std::chrono::microseconds current_time = GetTime<std::chrono::microseconds>();
+        if (cache->m_update_addr_response < current_time) {
```

Comment:
if you initialized `m_update_addr_response` to be 0 with `std::chrono::microseconds m_update_addr_response{0};`, wouldn't this clause always be true?

Reply:
It is true on the first ever iteration, but then `cache->m_update_addr_response = PoissonNextSend(current_time, AVG_ADDRESS_RESPONSE_UPDATE_INTERVAL);` executes and makes it non-zero right away.